### PR TITLE
Limit mount to root context-path #10546

### PR DESCRIPTION
**Pull request by anatol-sialitski**

### CODE REVIEW

File: modules/portal/portal-impl/src/main/java/com/enonic/xp/portal/impl/handler/SlashApiHandler.java [hunk 1]
```diff
@@ -55,11 +55,16 @@ public class SlashApiHandler
     private static final Pattern MOUNT_SITE_ENDPOINT_PATTERN =
         Pattern.compile( "^/(admin/)?site/(?<project>[^/]+)/(?<branch>[^/]+)(?<contentPath>/.*?)/_/" );
 
-    private static final Pattern MOUNT_WEBAPP_ENDPOINT_PATTERN = Pattern.compile( "^/webapp/(?<baseAppKey>[^/]+)(?<restPath>/.*?)/_/" );
+    private static final Pattern MOUNT_WEBAPP_ENDPOINT_PATTERN = Pattern.compile( "^/webapp/(?<baseAppKey>[^/]+)(?<restPath>.*?)/_/" );
 
     private static final List<String> RESERVED_APP_KEYS =
         List.of( "attachment", "image", "error", "idprovider", "service", "asset", "component", "widgets", "media" );
 
+    private static final String WEBAPP_DEFAULT_CONTEXT_PATH_PATTERN = "^/webapp/([^/]+)/_/.*";
```
Comment (rymsha) [inline modules/portal/portal-impl/src/main/java/com/enonic/xp/portal/impl/handler/SlashApiHandler.java:63]: Compile it, and give group a name

File: modules/portal/portal-impl/src/main/java/com/enonic/xp/portal/impl/handler/SlashApiHandler.java [hunk 2]
```diff
@@ -55,11 +55,16 @@ public class SlashApiHandler
     private static final Pattern MOUNT_SITE_ENDPOINT_PATTERN =
         Pattern.compile( "^/(admin/)?site/(?<project>[^/]+)/(?<branch>[^/]+)(?<contentPath>/.*?)/_/" );
```
Comment (rymsha) [inline modules/portal/portal-impl/src/main/java/com/enonic/xp/portal/impl/handler/SlashApiHandler.java:56]: How about project path? I'm not sure it is captured by this regex

File: modules/portal/portal-impl/src/main/java/com/enonic/xp/portal/impl/handler/SlashApiHandler.java [hunk 3]
```diff
@@ -213,12 +227,21 @@ private boolean verifyRequestMountedOnSites( final ApiDescriptor apiDescriptor,
...
+            final Project project = projectService.get( ProjectName.from( portalRequest.getRepositoryId() ) );
+            if ( project == null || project.getSiteConfigs().get( apiDescriptor.key().getApplicationKey() ) == null )
+            {
+                return false;
+            }
+        }
+
+        if ( apiDescriptor.getContextPath() == ApiContextPath.DEFAULT )
+        {
+            final String siteRelativePath = (String) ContextAccessor.current().getAttribute( SITE_RELATIVE_PATH_CTX_ATTRIBUTE );
```
Comment (rymsha) [inline modules/portal/portal-impl/src/main/java/com/enonic/xp/portal/impl/handler/SlashApiHandler.java:243]: If we only use it in one place it is better to propagate it via method parameters

**Review by rymsha — CHANGES_REQUESTED** [review]

File: modules/portal/portal-impl/src/main/java/com/enonic/xp/portal/impl/handler/SlashApiHandler.java (hunk 2 above)
Comment (anatol-sialitski) [reply modules/portal/portal-impl/src/main/java/com/enonic/xp/portal/impl/handler/SlashApiHandler.java:56]: We do not have requirements to support project path at least for now.

**Review by rymsha — APPROVED** [review]

**Pull request merged**